### PR TITLE
Http Client Processor module

### PR DESCRIPTION
**Pull request by whummer**
This is a simple implementation of a http client processor module, which aims to reproduce the semantics of `http-client` (in Spring XD) for SCDF.

Example (assuming the module is registered as "http-client"):

```
stream create --name foo --definition "time --fixedDelay=10 | http-client --url='https://httpbin.org/post' --httpMethod=POST --body='{\"foo\":\"bar\"}' | log" --deploy
```

The log output should contain the POSTed data:

```
{
  "url": "https://httpbin.org/post",
  "data": "\\{\\\"foo\\\":\\\"bar\\\"\\}", 
  ...
}
```

### CODE REVIEW

File: httpclient-processor/src/main/java/org/springframework/cloud/stream/module/http/HttpClientProcessor.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+			HttpHeaders headers = new HttpHeaders();
+			// TODO: add headers
+			HttpMethod method = HttpMethod.valueOf(properties.getHttpMethod());
+			URI uri = new URI(properties.getUrl());
+			HttpEntity<?> requestEntity = new RequestEntity<Object>(properties.getBody(), headers, method, uri);
+			ResponseEntity<String> httpResponse = restTemplate.exchange(
+					properties.getUrl(), 
+					HttpMethod.valueOf(properties.getHttpMethod()),
+					requestEntity,
+					String.class);
```
Comment (markfisher) [inline httpclient-processor/src/main/java/org/springframework/cloud/stream/module/http/HttpClientProcessor.java]: expectedReturnType could be a configurable property

**Comment by markfisher** [comment]
Thanks for the contribution! For some of the properties, we might want to consider supporting SpEL expressions that could be evaluated with the inbound Message as the root object for evaluation (I'm especially thinking of the request "body").

**Comment by whummer** [comment]
updated the PR:
- add SpEL expression support for URL, headers, and request body
- allow configuration of expectedReturnType
- extend tests to cover the new functionality
- updated README

File: httpclient-processor/src/main/java/org/springframework/cloud/stream/module/http/HttpClientProcessorProperties.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import org.springframework.expression.Expression;
+import org.springframework.expression.spel.standard.SpelExpressionParser;
+
+/**
+ * Configuration properties for the Http Client Processor module.
+ *
+ * @author Waldemar Hummer
+ */
+@ConfigurationProperties
+public class HttpClientProcessorProperties {
```
Comment (markfisher) [inline httpclient-processor/src/main/java/org/springframework/cloud/stream/module/http/HttpClientProcessorProperties.java:12]: I think for the expression-based properties we probably still do want to support simple literals (i.e. not requiring the user to wrap literals with quotes). For example, we should have mutually exclusive properties `body` and `bodyExpression`. Sorry that was not clear when I originally suggested relying on Expressions. I can make that change while merging, though.

Comment (whummer) [inline httpclient-processor/src/main/java/org/springframework/cloud/stream/module/http/HttpClientProcessorProperties.java:12]: ok great. Anything else I can help with the merge, please let me know.

**Comment by markfisher** [comment]
I'm reassigning this to @ericbottard, and he'll most likely replace this PR after making some additional changes to what I've added on top of this PR via this commit: https://github.com/markfisher/spring-cloud-stream-modules/commit/d909cc97b36db52b42b20de16dbc6c3706c7a680

**Comment by ericbottard** [comment]
Closing in favor of https://github.com/spring-cloud/spring-cloud-stream-modules/pull/60

**Pull request closed without merging**